jgrapht-core/src/main/java/org/jgrapht/alg/dual/LineGraphConstructor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * (C) Copyright 2018-2018, by Nikhil Sharma and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.dual;
+
+import org.jgrapht.Graph;
+
+/**
+ * Generator which produces the
+ * <a href="http://mathworld.wolfram.com/LineGraph.html">line graph</a> of a given input
+ * graph.
+ * The line graph of an undirected graph $G$ is another graph $L(G)$ that represents the
+ * adjacencies between edges of $G$.
+ * The line graph of a directed graph $G$ is the directed graph $L(G)$ whose vertex set
+ * corresponds to the arc set of $G$ and having an arc directed from an edge $e_1$ to an
+ * edge $e_2$ if in $G$, the head of $e_1$ meets the tail of $e_2$
+ *
+ * The runtime complexity is $O(|V|+|E|)$.
```

Comment:
This runtime complexity does not seem to be correct.

Reply:
Changed to O(V). Please correct me if I am wrong.